src/Mvc/test/Microsoft.AspNetCore.Mvc.FunctionalTests/SerializableErrorTests.cs

```diff
@@ -89,25 +77,50 @@ public async Task PostedSerializableError_IsBound(string acceptHeader)
             XmlAssert.Equal(expectedXml, responseData);
         }
 
-        [ConditionalTheory]
-        // Mono issue - https://github.com/aspnet/External/issues/18
-        // XmlSerializer test is disabled Mono.Xml2.XmlTextReader.ReadText is unable to read the XML.
-        // This is fixed in mono 4.3.0.
-        [FrameworkSkipCondition(RuntimeFrameworks.Mono)]
-        [InlineData("application/xml-xmlser")]
-        [InlineData("application/xml-dcs")]
+        [Theory]
+        [MemberData(nameof(AcceptHeadersData))]
         public async Task IsReturnedInExpectedFormat(string acceptHeader)
         {
             // Arrange
             var input = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
-                "<Employee xmlns=\"http://schemas.datacontract.org/2004/07/XmlFormattersWebSite.Models\">" +
+                "<Employee" +
+                (string.Equals("application/xml-dcs", acceptHeader) ?
+                    " xmlns =\"http://schemas.datacontract.org/2004/07/XmlFormattersWebSite.Models\"" :
+                    string.Empty) +
+                ">" +
                 "<Id>2</Id><Name>foo</Name></Employee>";
             var expected = "<Error><Id>The field Id must be between 10 and 100.</Id>" +
                 "<Name>The field Name must be a string or array type with a minimum " +
                 "length of '15'.</Name></Error>";
             var request = new HttpRequestMessage(HttpMethod.Post, "http://localhost/SerializableError/CreateEmployee");
             request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse(acceptHeader));
-            request.Content = new StringContent(input, Encoding.UTF8, "application/xml-dcs");
+            request.Content = new StringContent(input, Encoding.UTF8, acceptHeader);
```

Comment:
This test previously exercised only the DCS input formatter. And, now that it tests both input formatters, need to deal with different namespace expectations of the underlying serializers. (Web site does not override the default DCS namespace for the `Employee` type.)

Reply:
This is why theory tests make me sad.